pyvalet/interpreter.py

```diff
@@ -255,22 +411,37 @@ def _get_series_observations(self, series, response_format='csv', **kwargs):
             else:
                 n_series = 1
             # For passed single string.
-            response = self._get_observations(series, response_format=response_format, **kwargs)
-            df = self._pandafy_response(response.text, skiprows=4)  # TODO: This will not work with comma sep series.
-            df_series = df.iloc[0:n_series]
-            df = df.iloc[1+n_series:]
-            headers = df.iloc[0]
-            df = pd.DataFrame(df.values[1:], columns=headers)
-            if self.logger is not None:
-                self.logger.debug(f"The {series} series has {df.shape[0]} observations")
+            if response_format=='csv':
+                response = self._get_observations(series, response_format=response_format, **kwargs)
+                df = self._pandafy_response(response.text, skiprows=4)  # TODO: This will not work with comma sep series.
+                df_series = df.iloc[0:n_series]
```

Comment:
This could be clearer in the original code:

`df_series` is the dataframe containing information about the series. It has `rows == n_series` (so here, typically just 1). `df` becomes the observations after a bit of wrangling.

I bring this up because your `js_series` and `series` variables are returned in the opposite order, this caused a bug in `get_series_observations` when trying to use the old, csv method to get data. I'll follow up with a commit to solve this issue